common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -141,7 +154,15 @@ public static class Collation {
     public final boolean supportsLowercaseEquality;
 
     public Collation(
```

Comment:
I would say this would be better of as adding only two functions to CollationFactory, one to list all collations and the other to build collation info based on the `CollationIdentifier`. These two should be sufficient to make all this work and to keep code clean.